helper/winsvc/mock_windows_service.go

```diff
@@ -0,0 +1,297 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package winsvc
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/shoenig/test/must"
```

Comment:
yeah I don't remember where I picked up this opinion from (maybe I should refresh myself), but having prod code with exported test methods feels funny.  I think I'd feel slightly less funny if this was stashed in a sub-package? `helper/winsvc/mock`?

sorry this comment is so vibes-based without any real evidence on hand.

Reply:
I'm good with that :+1: